pkg/cmd/run/rerun/rerun.go

```diff
@@ -0,0 +1,121 @@
+package rerun
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/run/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type RerunOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	RunID string
+
+	Prompt bool
+}
+
+func NewCmdRerun(f *cmdutil.Factory, runF func(*RerunOptions) error) *cobra.Command {
+	opts := &RerunOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:    "rerun [<run-id>]",
+		Short:  "Rerun a given run",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if len(args) > 0 {
+				opts.RunID = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("run ID required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRerun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func runRerun(opts *RerunOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("failed to determine base repo: %w", err)
+	}
+
+	runID := opts.RunID
+
+	if opts.Prompt {
+		cs := opts.IO.ColorScheme()
+		runs, err := shared.GetRunsWithFilter(client, repo, 10, func(run shared.Run) bool {
+			if run.Status != shared.Completed {
+				return false
+			}
+			// TODO StartupFailure indiciates a bad yaml file; such runs can never be
+			// rerun. But hiding them from the prompt might confuse people?
+			return run.Conclusion != shared.Success && run.Conclusion != shared.StartupFailure
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get runs: %w", err)
+		}
+		if len(runs) == 0 {
+			return errors.New("no recent runs have failed; please specify a specific run ID")
+		}
+		runID, err = shared.PromptForRun(cs, runs)
```

Comment:
Not new in this PR, but: I still don't think that it's safe to use ANSI color sequences in Survey prompt options. https://github.com/cli/cli/pull/2923#discussion_r592608026

Reply:
oops, forgot to change that in other places.

Reply:
oh this is already done. where are you seeing ANSI color codes going into survey?

`PromptForRun` does not use any color.